parsers/ES.py

```diff
@@ -1,27 +1,87 @@
 #!/usr/bin/env python3
 
 from datetime import datetime, timedelta, timezone
+from json import loads
 from logging import Logger, getLogger
 from typing import Callable, Dict, List, Optional
 
-# package "ree" is used to parse data from www.ree.es
-# maintained on github by @hectorespert at https://github.com/hectorespert/ree
-from ree import (
-    Ceuta,
-    ElHierro,
-    Formentera,
-    Gomera,
-    GranCanaria,
-    IberianPeninsula,
-    Ibiza,
-    LanzaroteFuerteventura,
-    LaPalma,
-    Mallorca,
-    Melilla,
-    Menorca,
-    Response,
-    Tenerife,
-)
+from arrow import get, utcnow
+
+FUEL_MAPPING = {
+    "dem": "demand",
+    "nuc": "nuclear",
+    "die": "oil",
+    "genAux": "oil",
+    "gas": "gas",
+    "gf": "gas",
+    "eol": "wind",
+    "cc": "gas",
+    "vap": "oil",
+    "fot": "solar",
+    "sol": "solar",
+    "hid": "hydro",
+    "car": "coal",
+    "resid": "biomass",
+    "termRenov": "unknown",
+    "cogenResto": "unknown",
+    "cogen": "gas",
+}
+
+LINK_MAPPING = {
+    "cb": "pe_ma",
+    "icb": "pe_ma",
+    "emm": "ma_me",
+    "emi": "ma_ib",
+    "eif": "ib_fo",
+    "inter": "int",
+}
+
+API_CODE_MAPPING = {
+    "IberianPeninsula": "DEMANDAQH",
+    "Ceuta": "CEUTA5M",
+    "Melilla": "MELILLA5M",
+    "Mallorca": "MALLORCA5M",
+    "Menorca": "MENORCA5M",
+    "Ibiza": "IBIZA5M",
+    "Formentera": "FORMENTERA5M",
+    "GranCanaria": "GCANARIA5M",
+    "Gomera": "LA_GOMERA5M",
+    "LaPalma": "LA_PALMA5M",
+    "Tenerife": "TENERIFE5M",
+    "LanzaroteFuerteventura": "LZ_FV5M",
+    "ElHierro": "EL_HIERRO5M",
+}
+
+ZONE_MAPPING = {
+    "IberianPeninsula": "Peninsula",
+    "Ceuta": "Peninsula",
+    "Melilla": "Peninsula",
+    "Mallorca": "Baleares",
+    "Menorca": "Baleares",
+    "Ibiza": "Baleares",
+    "Formentera": "Baleares",
+    "GranCanaria": "Canarias",
+    "Gomera": "Canarias",
+    "LaPalma": "Canarias",
+    "Tenerife": "Canarias",
+    "LanzaroteFuerteventura": "Canarias",
+    "ElHierro": "Canarias",
+}
```

Comment:
These mappings should be using the zone keys instead of the names. This also includes the timezones.